Added utilities directories with ImportanceSample.py and ProjectionTool.py

serenavinciguerra89 opened the pull request.

thjsal commented:
Can you add  `return ax` at the end of the Projection tool code? Then the users can further modify the figure if they want.

serenavinciguerra89 commented:
Do you see the added return ax now? or do I have to Tell Git that I want a pull request with that commit in as well?

sguillot commented:
We see it :-)

thjsal commented:
At some point, you also created an option to have an antipodal projection of one of the spots. Do you think that could be added to this version as well?

serenavinciguerra89 commented:
I had completely forgotten about that, sorry…
I need to check how I did that and if it is easy to transfer it, I’ll do it
(I need to check if I modified the file directly without having put it as
an option or if I added the option)
I’ll check that tomorrow.

On Sat 1. Oct 2022 at 15:23, Tuomo Salmi ***@***.***> wrote:

> At some point, you also created an option to have an antipodal projection
> of one of the spots. Do you think that could be added to this version as
> well?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/xpsi-group/xpsi/pull/145#issuecomment-1264365229>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEUHPDPE7GVQNXTPXYRYVMTWBA3O3ANCNFSM6AAAAAAQTDRQYE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

thjsal commented:
No worries. Now I also realize this version we still have "Nord pole" instead of "North pole" :). I suggest to fix that as well. 

> I had completely forgotten about that, sorry… I need to check how I did that and if it is easy to transfer it, I’ll do it (I need to check if I modified the file directly without having put it as an option or if I added the option) I’ll check that tomorrow.
> […](#)
> On Sat 1. Oct 2022 at 15:23, Tuomo Salmi ***@***.***> wrote: At some point, you also created an option to have an antipodal projection of one of the spots. Do you think that could be added to this version as well? — Reply to this email directly, view it on GitHub <[#145 (comment)](https://github.com/xpsi-group/xpsi/pull/145#issuecomment-1264365229)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AEUHPDPE7GVQNXTPXYRYVMTWBA3O3ANCNFSM6AAAAAAQTDRQYE> . You are receiving this because you authored the thread.Message ID: ***@***.***>

serenavinciguerra89 commented:
I cannot see the indentation from the github (I thought I solved those on
Wednesday and now in my version I checked and the indentations of those
lines seem fine). Added the __init__.py and the line in the setup.py

Il giorno ven 14 ott 2022 alle ore 23:35 Tuomo Salmi <
***@***.***> ha scritto:

> I have checked that the ImportanceSample.py script should work now,
> although when upgrading to Python3 , at least all raw_input need to be
> changed with input.
>
> For ProjectionTool there are still a couple of indentation errors to be
> fixed (where if antipodal:).
>
> To be able to actually import ProjectionTool directly from X-PSI, need to
> also add en empty __init__.py file inside utilities folder and
>
> -                    'xpsi/pixelmesh'],
> +                    'xpsi/pixelmesh',
> +                    'xpsi/utilities'],
>
> in setup.py.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/xpsi-group/xpsi/pull/145#issuecomment-1279492173>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEUHPDJEI2B3PGERCFCVLHLWDHG27ANCNFSM6AAAAAAQTDRQYE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

thjsal commented on xpsi/utilities/ProjectionTool.py:
```diff
@@ -0,0 +1,648 @@
+                j = j+1
+            CAall.append(coli)
+            
+    if antipodal:
```
I think here is at least an indentation error. There should be 4 more spaces before `if antipodal`.

serenavinciguerra89 commented:
They do not seem to have a weird indent on my version (also in the code
lines you printed it is not clear if it has a problem with indent as I
think it should align with the else above those lines... see attachment).
I also tested it in one post-processing and I could import the file without
problems and use it normally. I also checked the antipodal option, and it
works as far as I can tell.


Il giorno sab 15 ott 2022 alle ore 19:06 Tuomo Salmi <
***@***.***> ha scritto:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In xpsi/utilities/ProjectionTool.py
> <https://github.com/xpsi-group/xpsi/pull/145#discussion_r996330840>:
>
> > +        TAall = TA_p+TA_s
> +        LABall = labels_p +labels_s
> +        ind = [1,2,4,5] if len(TAall)>2 else [0,6]
> +
> +        argT = np.argsort(TAall)
> +        j = 0
> +        for i in range(len(LABall)):
> +            if (OMIlab in LABall[i]):
> +                coli = 'black'
> +            else:
> +                ind_i = ind[argT[j]]
> +                coli = mycolors0[ind_i]
> +                j = j+1
> +            CAall.append(coli)
> +
> +    if antipodal:
>
> I think here is at least an indentation error. There should be 4 more
> spaces before if antipodal.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/xpsi-group/xpsi/pull/145#pullrequestreview-1143208675>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEUHPDKC472Z6MF2B6YHNBTWDLQBNANCNFSM6AAAAAAQTDRQYE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

thjsal commented:
I don't see any attachments But in case that `if antipodal` is aligned correctly, the problem is that the lines following it are double intended (they are 8 spaces more intended than `if antipodal` at the moment, right? Or is your version the same as currently in GitHub?

thjsal commented:
To summarize: The indentation problem above was solved, bu the CI test still fails, since it apparently tries to read the Projection Tool (for some reason) as if that would need to be written in Python3 (although that is not yet required).  
Regardless of that, I think we could proceed to merge this. However, I think we should update the CHANGELOG-file with this, and update X-PSI version numbers to e.g. 1.0.1.
